hallow_test.go

```diff
@@ -31,3 +32,41 @@ func TestCreatePrincipalName(t *testing.T) {
 		})
 	}
 }
+
+func TestValidatePublicKey(t *testing.T) {
+	h := config{}
+
+	for _, c := range []struct {
+		pubKey      string
+		expectedErr error
+	}{
+		// Valid keys.
+		{
+			pubKey: "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIJOfreF0kMkdJ1ISFvPsucJ7X8UJ07rQV99hQGLYBuSV",
+		},
+		{
+			pubKey: "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBKEKlDRLGPX1iKEGebvigYpkydGfuok6WQYROznG8XFyH1Se7/p1pXADdJtrnegU2Qn3jgmevHvDKD5VAIyGpB8=",
+		},
+		{
+			pubKey: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQCn9+7X+Uk8NSi7Dv6BRkKLVKz6MAKvX/jYhY7Lq/KfxZGlSPcc4duhisKR+YN7Sooo4xXURNUakzuDqptBZVgjqip7iedAqIlDmwTQsoYW+IVsrDg3T/3SUgAy3CsPcMFEE/S13o/haXa9iHZZl2TEGy7qIY7llUUI33SV/EZ3JBcznnLUqFVKlfjB5pKUlixkl+aMFOTvHUrwWv4UtOs0oczvrFQLihLj34Wqeq9LpN/6lXtVppzcxcAX3JGvmUx1GM4MJA9wXgex/1SXEvwEgkQjd+bSsdPazAV8+1dAqvbPgbyqSJmeCJ1biu/1FaBB4Zni0Dwjl0Rq/IVJ0Z+V8KUgZvQCMQDmRBxKBfkmIR7zlhR66ZYt/FC3L/QDAM6JAKDGh+dURWuI8h75w8OwCItRIEAvm6w8Rjzbxdjb7DOjA/toiyJBgmLOb2ZgkldtpKM+3VrPBzvPl7tSHPK0K01E4z9huIRPw0/3cvaZxMbJshkwcOok33msKRl7MwdWxI3yhIyoElWzNH8RsRl/171PPh/bxWDAd43uX/YfFkY5h0XHvnPUK2BMWFkJps1zLqrA+qFfQAJxHWLDl0keLaCVmpQs0PG+Q+NsXlSqGNSTN/r6Yz4s/qn0A9aJ651iVcnbusdnBmF/sCKPa0WD78nXyD5uFtbZIyzU0xjq6w==",
+		},
+		// DSA key
+		{
+			pubKey:      "ssh-dss AAAAB3NzaC1kc3MAAACBANLlxcoOBh5rcRm3b0hg7kN31pFFesd4rAncMPe230bNabgjqCEblZPyCkqP9D4aktKIqiCk43YjreXkDrB/1a1ST0ZjGu4914eGIW68W1vCtqOqFe21kHWLhh/HhZXlWHLwrHu9RkcMOAghDhj/tlkmgu09WfTnJnuXKqrAIYmnAAAAFQCJjFbwiJP976BeSCX3tNLFzR5JHwAAAIEArAdfNtpmnThMD6guamSKg17vv1MtFCxg7xuP7kweFPFepzD+l/xKXsUq1nnTRFqF4HDsHT0xgXY5567wBfQEqATFBxY7Zd/8298TY8aQbLcjLr+pQ9bQRMjKM2XOjVr31neNSJf51DaCjmvNWMv5vnCBoIDXY72TJvSryIN/W9MAAACAFrPrlKRD746a/Qr0+ZOyUI4GJC0e04zgG9a/tLNh6cNyBn6nVVgyCOLhQqONyhZks4ZUFTHphNpEUGBUgN8Ox4kaYf4wQB6G+SvcprjZrC06RCQGJYS5vFSgNpqrh/6nCAaeDtsFH3Lx5ot/sxQYw2OzTOdkbSRCBV/SNBruDNI=",
+			expectedErr: unknownKeyTypeError,
```

Comment:
In the future I may refactor this to be more explicitly a bad key, not just that we don't know about it, but this is fine for now